### PR TITLE
Only use `temp` dependency, not `tmp-sync`

### PR DESCRIPTION
**Pull request by akatov**
Don't use `tmp-sync` dependency which eventually depends on a deprecated `lodash` version.
Instead, use promise-wrapped functions from `temp`

### CODE REVIEW

File: lib/utilities/mk-tmp-dir-in.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+var Promise   = require('../ext/promise');
+var fs        = require('fs-extra');
+var temp      = require('temp');
+var mkdir     = Promise.denodeify(fs.mkdir);
+var mkdirTemp = Promise.denodeify(temp.mkdir);
+
+function exists(dir) {
+  return new Promise(function(resolve) {
+    fs.exists(dir, resolve);
```
Comment (stefanpenner) [inline lib/utilities/mk-tmp-dir-in.js]: Can't this also be denodeify?

File: lib/utilities/mk-tmp-dir-in.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  });
+}
+
+function mkTmpDirIn(dir) {
+  return exists(dir).then(function(doesExist) {
+    if (!doesExist) {
+      mkdir(dir);
```
Comment (stefanpenner) [inline lib/utilities/mk-tmp-dir-in.js]: Missing return

File: lib/utilities/mk-tmp-dir-in.js (hunk 1 above)
Comment (akatov) [inline lib/utilities/mk-tmp-dir-in.js]: `fs.exists` has a different signature - the callback receives only one argument - the boolean value.

**Comment by stefanpenner** [comment]
@akatov good point

File: lib/utilities/mk-tmp-dir-in.js (hunk 2 above)
Comment (akatov) [inline lib/utilities/mk-tmp-dir-in.js]: hmm, we aren't using the return value of the promise, nor does the promise actually return a value..
however, if mkdir rejects we want the containing promise to reject also. So I'll update this.

**Comment by akatov** [comment]
I was tempted to include normalize/mz but didn't want to add an extra dependency...

**Comment by stefanpenner** [comment]
@homu r+

**Comment by stefanpenner** [comment]
@homu retry

**Pull request merged**